smithy-cli/src/main/java/software/amazon/smithy/cli/dependencies/FileCacheResolver.java

```diff
@@ -113,17 +113,28 @@ private List<ResolvedArtifact> load() {
             throw new DependencyResolverException("Error loading dependency cache file from " + location, e);
         }
 
-        List<ResolvedArtifact> result = new ArrayList<>(node.getStringMap().size());
-        for (Map.Entry<String, Node> entry : node.getStringMap().entrySet()) {
-            Path artifactLocation = Paths.get(entry.getValue().expectStringNode().getValue());
-            long lastModifiedOfArtifact = artifactLocation.toFile().lastModified();
+        // If the version of the cache file does not match the current version or does not exist
+        // invalidate it so we can replace it with a more recent version.
+        if (!node.containsMember("version")
+                || !node.expectStringMember("version").getValue().equals(CURRENT_CACHE_FILE_VERSION)
```

Comment:
```suggestion
                || !CURRENT_CACHE_FILE_VERSION.equals(node.expectStringMember("version").getValue())
```

Rationale, we know that `CURRENT_CACHE_FILE_VERSION` will never be `null`, thus this way the code is less prone to null pointer exceptions.